src/hb-machinery.hh

```diff
@@ -686,13 +689,22 @@ struct BEInt<Type, 2>
   typedef Type type;
   inline void set (Type V)
   {
+#if defined(__x86_64) || defined(__ARM_NEON__) || defined(__ARM_NEON)
+    uint16_t *ptr = (uint16_t *)v;
+    *ptr = bswap_16(V);
+#else
     v[0] = (V >>  8) & 0xFF;
     v[1] = (V      ) & 0xFF;
+#endif
   }
   inline operator Type (void) const
   {
+#if defined(__x86_64) || defined(__ARM_NEON__) || defined(__ARM_NEON)
+    return bswap_16(*(uint16_t *)this);
+#else
```

Comment:
We typically prefer to add return to the `#ifdef`'d code and instead of using `#else`, leave the defualt iplementation always compiled, just unreached.